src/index.js

```diff
@@ -60,9 +60,7 @@ export default class Machine extends erx.Bus<State> {
       this.transitioning = true;
       p(tFn(this, data)).then((change) => {
         this.transitioning = false;
-        if (change) {
-          finish(next);
-        }
+        change && finish(next);
```

Comment:
Isn't this still gonna get weird if change is promise?

Reply:
nop, p(tFn()) will always return a promise so change will be a value .then ...